src/layers/legacy/medCoreLegacy/parameters/medAbstractParameterL.cpp

```diff
@@ -26,8 +26,6 @@ class medAbstractParameterLPrivate
     QString toolTip;
 
     bool hide;
-
-    ~medAbstractParameterLPrivate() {delete label;}
```

Comment:
The label is created in this class and used through `getLabel`, but there is no guarantee that it has been retrieved and added to another Qt object (that would take care of its deletion). Therefore to avoid a memory leak it would be better to keep the delete but only do it if `label->parent()` is null.

Reply:
I am trying several idea including yours and for now i can't avoid a crash here or mem leak. This is not particularly here but the way 4D parameters are used

Reply:
I added a commit. With that, i did not see mem leaks with valgrind here